plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightService.java

```diff
@@ -0,0 +1,171 @@
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.common.transport.BoundTransportAddress;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.SecureTransportSettingsProvider;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.client.Client;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Objects;
+
+/**
+ * FlightService manages the Arrow Flight server and client for OpenSearch.
+ * It handles the initialization, startup, and shutdown of the Flight server and client,
+ * as well as managing the stream operations through a FlightStreamManager.
+ */
+public class FlightService extends NetworkPlugin.AuxTransport {
+    private static final Logger logger = LogManager.getLogger(FlightService.class);
+    private final ServerComponents serverComponents;
+    private StreamManager streamManager;
+    private Client client;
+    private FlightClientManager clientManager;
+    private SecureTransportSettingsProvider secureTransportSettingsProvider;
+    private BufferAllocator allocator;
+    private ThreadPool threadPool;
+
+    /**
+     * Constructor for FlightService.
+     * @param settings The settings for the FlightService.
+     */
+    public FlightService(Settings settings) {
+        Objects.requireNonNull(settings, "Settings cannot be null");
+        try {
+            ServerConfig.init(settings);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to initialize Arrow Flight server", e);
+        }
+        this.serverComponents = new ServerComponents(settings);
+    }
+
+    void setClusterService(ClusterService clusterService) {
+        serverComponents.setClusterService(Objects.requireNonNull(clusterService, "ClusterService cannot be null"));
+    }
+
+    void setNetworkService(NetworkService networkService) {
+        serverComponents.setNetworkService(Objects.requireNonNull(networkService, "NetworkService cannot be null"));
+    }
+
+    void setThreadPool(ThreadPool threadPool) {
+        this.threadPool = Objects.requireNonNull(threadPool, "ThreadPool cannot be null");
+        serverComponents.setThreadPool(threadPool);
+    }
+
+    void setClient(Client client) {
+        this.client = client;
+    }
+
+    void setSecureTransportSettingsProvider(SecureTransportSettingsProvider secureTransportSettingsProvider) {
+        this.secureTransportSettingsProvider = secureTransportSettingsProvider;
+    }
+
+    /**
+     * Starts the FlightService by initializing the stream manager.
+     */
+    @SuppressWarnings("removal")
+    @Override
+    protected void doStart() {
+        try {
+            allocator = AccessController.doPrivileged((PrivilegedAction<BufferAllocator>) () -> new RootAllocator(Integer.MAX_VALUE));
+            serverComponents.setAllocator(allocator);
+            SslContextProvider sslContextProvider = ServerConfig.isSslEnabled()
```

Comment:
Why do we need `DisabledSslContextProvider` ? If there is no secure transport configured, we just shouldn't set any SSL context at all

Reply:
let me try to refactor.

Reply:
removed